CODING_GUIDELINES.md

```diff
@@ -51,6 +51,44 @@ We deviate from the [Google Python Style Guide][google-style-guide] only in the
   - Client code (like tests, doctests and examples) should use the above style for public FieldView API
   - Library code should always import the defining module and use qualified names.
 
+### Error messages
+
+Error messages should be written as sentences, starting with a capital letter and ending with a period (avoid exclamation marks). Try to be informative without being verbose. Code object like 'ClassNames' and 'function_names' etc should be single-quoted and so should string values used to interpolate the message.
```

Comment:
Is this common practice to use full sentence? @petiaccja pointed out in #1292 that Python does formatting different. What does numpy and other big libraries do?  Also not explicitly stated, the google style guide implicitly (by examples) suggests what you propose.

Reply:
I tried to find the reasoning behind the Python builtins convention but there doesn't even seem to be a conscious decision on record (at least not easily findable).